main.go

```diff
@@ -2,9 +2,41 @@ package main
 
 import (
 	"os"
+  "os/signal"
 )
 
 func main() {
+	var test string
+	exitsig := make(chan os.Signal, 1)
+	signal.Notify(exitsig, os.Interrupt)
+	var (
+			lockstate bool = false
+	)
```

Comment:
Since you're declaring only one variable, I think it's better to just initialize it the sort way:

```go
lockstate := false
```